storage/innobase/lock/lock0wait.cc

```diff
@@ -196,10 +196,16 @@ static void print_lock_wait_timeout(const trx_t &trx,
   size_t unused;
 
   outs << "Lock wait timeout info:\n";
-  outs << "Requested thread id: " << thd_get_thread_id(trx.mysql_thd) << "\n";
+  if (trx.mysql_thd)
+    outs << "Requested thread id: " << thd_get_thread_id(trx.mysql_thd) << "\n";
```

Comment:
IMHO outs << "foo :" << (trx.mysql_thd ? thd_get_thread_id() : "<background>" << "\n" would be very slightly better. No need to make this change unless you feel like it. Same comment below

Reply:
Yeah, it's not quite as simple as that, placing conditional within a stream...it is a runtime decision that changes types to operator<<(...) which is needed to be known at compile time.  i.e.  thd_get_thread_id() returns some kind of integer and would call operator<<(int) or something whereas "" would call operator<<(const char*).